### PR TITLE
Add DefaultPulse device

### PR DESCRIPTION
**Pull request by lillian542**
Set up dummy device (currently default.qubit with addition of `dt`, `dim` and `drift` attributes) as first step in a device to support pulse programming

### CODE REVIEW

**Comment by lillian542** [comment]
[sc-31947]

**Review by Qottmann — APPROVED** [review]
Thanks for this @lillian542 💯

File: pennylane/devices/default_pulse.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        wires (int, Iterable[Number, str]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers (i.e., ``[-1, 0, 2]``)
+            or strings (``['ancilla', 'q1', 'q2']``). Default 1 if not specified.
+        shots (None, int): How many times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Defaults to ``None`` if not specified, which means that the device
+            returns analytical results.
+        dt (float): the time step used by the differential equation solver to evolve the
+             time-dependent Hamiltonian.
+        dim (int): dimensions of the system (2 for qubits, 3 for qutrits, etc)
+        drift(Operator): optional Hamiltonian term representing a constant drift term for the system Hamiltonian
```
Comment (trbromley) [inline pennylane/devices/default_pulse.py:37]: Should we perhaps omit these terms for now, until we're ready to use them?

**Comment by lillian542** [comment]
[sc-31947]

File: pennylane/devices/default_pulse.py (hunk 1 above)
Comment (AlbertMitjans) [reply pennylane/devices/default_pulse.py:37]: It would be also nice to raise a warning when drift is not None, saying that this currently does nothing

**Comment by Qottmann** [comment]
Just fyi, @trbromley and I discussed today that for the v0.29.0 we would actually not promote using `default.pulse`. this is mainly because it is still unclear whether providing the option for an omnipresent drift term makes sense or not. This device will be relevant however as soon as we move towards qutrits / qudtis. If its already ready to merge, we can do that, else I would put it on the backlog for the moment.

File: pennylane/devices/default_pulse.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+The default.pulse device builds on PennyLane's standard qubit-based device, adding support for
+pulse-based circuits.
+"""
+import numpy as np
+
+from .._version import __version__
+from .default_qubit import DefaultQubit
+
+
+class DefaultPulse(DefaultQubit):
```
Comment (lillian542) [inline pennylane/devices/default_pulse.py:24]:
```suggestion
class DefaultPulse(DefaultQubitJax):
```

File: pennylane/devices/default_pulse.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+The default.pulse device builds on PennyLane's standard qubit-based device, adding support for
+pulse-based circuits.
+"""
+import numpy as np
+
+from .._version import __version__
+from .default_qubit import DefaultQubit
```
Comment (lillian542) [inline pennylane/devices/default_pulse.py:21]:
```suggestion
from .default_qubit import DefaultQubitJax
```

File: pennylane/devices/default_pulse.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    def __init__(self, wires, *, shots=None, analytic=None, dt=1e-5, dim=2, drift=None):
+        r_dtype = np.float64
+        c_dtype = np.complex128
+
+        self.dt = dt
+        self.dim = dim
+        self.drift = drift
+
+        super().__init__(wires, shots=shots, r_dtype=r_dtype, c_dtype=c_dtype, analytic=analytic)
```
Comment (lillian542) [inline pennylane/devices/default_pulse.py:54]:
```suggestion
        super().__init__(wires, shots=shots, analytic=analytic)
```

File: pennylane/devices/default_pulse.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    """
+
+    name = "Default PennyLane plugin for pulses"
+    short_name = "default.pulse"
+    pennylane_requires = __version__
+    version = __version__
+    author = "Xanadu Inc."
+
+    def __init__(self, wires, *, shots=None, analytic=None, dt=1e-5, dim=2, drift=None):
+        r_dtype = np.float64
```
Comment (lillian542) [inline pennylane/devices/default_pulse.py:47]:
```suggestion
```

File: pennylane/devices/default_pulse.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        c_dtype = np.complex128
```
Comment (lillian542) [inline pennylane/devices/default_pulse.py:48]:
```suggestion
```

File: pennylane/devices/default_pulse.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+The default.pulse device builds on PennyLane's standard qubit-based device, adding support for
+pulse-based circuits.
+"""
+import numpy as np
```
Comment (lillian542) [inline pennylane/devices/default_pulse.py:18]:
```suggestion
```

**Comment by lillian542** [comment]
will use default.qubit.jax instead

**Pull request closed without merging**